src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -67,6 +67,15 @@ class DefaultSource
       throw new Exception("Escape character cannot be more than one character.")
     }
 
+    val comment = parameters.getOrElse("comment", "#")
+    val commentChar: Character = if (comment == null) {
```

Comment:
No, I don't think. We'll default to `#` only if the user didn't set a value for `comment` at all. If they've explicitly set it to `null`, that will still return `null`:

```scala
scala> Map("comment" -> null).getOrElse("comment", "#")
res1: String = null
```